### ISSUE TITLE
fileupload: support multiple files in simple mode

### ISSUE THREAD

**Issue by mertsincan** [primefaces/primefaces#4038]
Reported By PRO User;
> We are using p:fileupload component with simple mode.
```
<p:fileUpload mode="simple" multiple="true" value="#{fileUploadView.file}"/>.
<p:commandButton value="Submit" ajax="false" actionListener="#{fileUploadView.upload}" />
```
> Our requirement is to use the simple fileupload and upload multiple files. 
> After setting multiple="true", we are able to select multiple files but clicking on upload button, only one file is uploaded.

Added UploadedFile.getFileNames() method to access uploaded files. Exp; file.getFileNames();

**Comment by Rapster** [comment]
Technical debts added, see comments: https://sonarcloud.io/component_measures?id=org.primefaces%3Aprimefaces&metric=new_technical_debt

**Comment by cnsgithub** [comment]
@mertsincan @Rapster Added another commit comment here: https://github.com/primefaces/primefaces/commit/ad123a52c29f263183a72aae746ac921ec45abce#r31276355

**Comment by cnsgithub** [comment]
Sorry for blaming once again, but I cannot agree with changes made in this commit since it's very poor design.

Please have a look at `NativeUploadedFile `class, it needs some refactoring to properly support multiple files. 

Depending on the CTOR being called we either have the single vs. multiple file upload situation. Since there are two separate field members, `part `and `parts`, the other will stay `null` and therefore is predestinated for NPEs thrown later on. E.g. methods `getSize` or `getInputstream `are suitable for the single file upload case only.

@Rapster @tandraschko What do you think?

**Comment by Rapster** [comment]
I definitely agree with your analysis. We'll have to think of a better design... Whenever I have some time, I'll try to propose something

**Comment by Rapster** [comment]
BTW, in such definition (e.g mode simple + multiple true) only one filename is displayed. Would be nice to fix this before 6.3
